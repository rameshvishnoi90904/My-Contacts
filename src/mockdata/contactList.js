export const contactList = [
    {
      recordID: '1',
      givenName: 'Elon',
      familyName: 'Musk',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '2',
      givenName: 'Jack',
      familyName: 'Ma',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '3',
      givenName: 'Larry',
      familyName: 'Page',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '4',
      givenName: 'Mark',
      familyName: 'Zuck',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '5',
      givenName: 'Jeff',
      familyName: 'Bezos',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
];