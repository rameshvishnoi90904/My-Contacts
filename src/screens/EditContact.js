import React, {useState, useContext} from 'react';
import {
  Button,
  Platform,
  StyleSheet,
  KeyboardAvoidingView,
  TextInput,
  View,
  Text,
  ScrollView
} from 'react-native';
import Contacts from 'react-native-contacts';
import Icon from 'react-native-vector-icons/Ionicons';

import LoadingSpinner from '../components/LoadingSpinner';
import { UserContext } from '../context/context';

const EditContact = ({navigation, route}) => {
    const [userState, updateState] = useContext(UserContext);
    const toEditUser = route.params?.contactInfo;

    const [firstName, setFirstName] = useState(toEditUser ? toEditUser.givenName: "Mark")
    const [lastName, setLastName] = useState(toEditUser ? toEditUser.familyName: "Zuck")
    const [email, setEmail] = useState(toEditUser && toEditUser.emailAddresses.length > 0 ? toEditUser.emailAddresses[0].email: "rv@rv.com")
    const [phoneNumber, setPhoneNumber] = useState(toEditUser && toEditUser.phoneNumbers.length > 0 ? toEditUser.phoneNumbers[0].number: "8989898989")
    const [isLoading, setIsLoading] = useState(false);

    const saveContact = () => {
        if (toEditUser) {
            setIsLoading(true);
            const udpatedContact = {
                ...toEditUser,
                familyName: lastName,
                givenName: firstName,
                emailAddresses: [{
                    label: "work",
                    email: email,
                }],
                phoneNumbers: [{
                    label: 'mobile',
                    number: phoneNumber,
                }],
            }
            Contacts.updateContact(udpatedContact)
            updateState({type: 'updateContact',data: {toUpdate: udpatedContact}})
            setIsLoading(false);
            navigation.navigate('Home');
        } else {
            var newPerson = {
                emailAddresses: [{
                    label: "work",
                    email: email,
                }],
                familyName: lastName,
                givenName: firstName,
                phoneNumbers: [{
                    label: 'mobile',
                    number: phoneNumber,
                }],
            }
            setIsLoading(true);
            Contacts.addContact(newPerson).then((response) => {
                updateState({type: 'addNewContact',data: {contactData: response}})
                setIsLoading(false);
                navigation.goBack();
            });
        }
    }

    return (
        <KeyboardAvoidingView 
        behavior={Platform.OS == "ios" ? "padding" : "height"}
        keyboardVerticalOffset={100}
        style={{flex: 1}}>
            <ScrollView style={styles.container}>
                <View style={{alignItems: 'center', marginBottom: 32}}>
                    <View>
                        <Icon name="camera" size={58} style={{backgroundColor: '#8D8D8D', padding: 20, borderRadius: 100}} color={"white"}/>
                    </View>
                </View>
                <Text style={styles.labelText}>First name</Text>
                <TextInput 
                    style={styles.inputField} 
                    onChangeText={setFirstName}
                    value={firstName}
                    placeholder="First name"></TextInput>

                <Text style={styles.labelText}>Last name</Text>
                <TextInput 
                    style={styles.inputField} 
                    value={lastName}
                    onChangeText={setLastName}
                    placeholder="Last name"></TextInput>

                <Text style={styles.labelText}>Phone number</Text>
                <TextInput 
                    style={styles.inputField} 
                    onChangeText={setPhoneNumber}
                    value={phoneNumber}
                    keyboardType={'number-pad'}
                    placeholder="Phone"></TextInput>

                <Text style={styles.labelText}>Email</Text>
                <TextInput 
                    style={styles.inputField} 
                    onChangeText={setEmail}
                    value={email}
                    keyboardType={'email-address'}
                    placeholder="Email"></TextInput>
                <Button 
                    title="Save" 
                    onPress={saveContact}
                    accessibilityLabel="Save contact"
                    accessibilityHint="Save Contact detail and navigate back to home screen"
                />
            </ScrollView>
            {isLoading && <LoadingSpinner/>}
        </KeyboardAvoidingView>
    )
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
      paddingHorizontal: 32,
      paddingTop: 16,
      backgroundColor: 'white'
    },
    inputField: {
        borderWidth: 1,
        height: 58,
        paddingHorizontal: 12,
        borderRadius: 8,
        marginBottom: 16,
        borderColor: '#8D8D8D',
        fontSize: 16
    },
    profilePicture: {
        width: 158,
        height: 158,
        padding: 20,
    },
    labelText: {
        marginBottom: 8
    }
  });


export default EditContact;