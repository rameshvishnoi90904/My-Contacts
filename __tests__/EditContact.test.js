import { expect } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";
import EditContact from '../src/screens/EditContact';
import {contactList} from '../src/mockData/contactList';

describe("<EditContact />", () => {
    const mockData = contactList[0];
    it('renders correctly across screens', () => {
        const container = renderer.create(<EditContact route={{params: {contactInfo: mockData}}}/>)
        const tree = container.toJSON();
        expect(tree).toMatchSnapshot();
    });
});