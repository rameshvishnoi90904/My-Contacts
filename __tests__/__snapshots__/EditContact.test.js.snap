// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditContact /> renders correctly across screens 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "paddingHorizontal": 32,
        "paddingTop": 16,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 32,
          }
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 58,
                },
                Object {
                  "backgroundColor": "#8D8D8D",
                  "borderRadius": 100,
                  "padding": 20,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï†º
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        First name
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="First name"
        style={
          Object {
            "borderColor": "#8D8D8D",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 58,
            "marginBottom": 16,
            "paddingHorizontal": 12,
          }
        }
        value="Elon"
      />
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Last name
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Last name"
        style={
          Object {
            "borderColor": "#8D8D8D",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 58,
            "marginBottom": 16,
            "paddingHorizontal": 12,
          }
        }
        value="Musk"
      />
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Phone number
      </Text>
      <TextInput
        keyboardType="number-pad"
        onChangeText={[Function]}
        placeholder="Phone"
        style={
          Object {
            "borderColor": "#8D8D8D",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 58,
            "marginBottom": 16,
            "paddingHorizontal": 12,
          }
        }
        value="+23493023902"
      />
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Email
      </Text>
      <TextInput
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email"
        style={
          Object {
            "borderColor": "#8D8D8D",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 58,
            "marginBottom": 16,
            "paddingHorizontal": 12,
          }
        }
        value="andre@gmail.com"
      />
      <View
        accessibilityLabel="Save contact"
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
